### PR TITLE
Add sosreport and supportconfig modules

### PR DESCRIPTION
**Pull request by gregbdunn**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Adds modules that gathers sosreport and supportconfig outputs

### CODE REVIEW

File: mod.d/supportconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# language governing permissions and limitations under the License.
+
+--- !ec2rlcore.module.Module
+# Module document. Translates directly into an almost-complete Module object
+name: !!str supportconfig
+path: !!str
+version: !!str 1.0
+title: !!str Gather a supportconfig
+helptext: !!str |
+  Gather an sosreport
```
Comment (Drudenhaus) [inline mod.d/supportconfig.yaml:22]: s/sosreport/supportconfig

File: mod.d/supportconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  trap error_trap ERR
+
+  # read-in shared function
+  source functions.bash
+
+  echo "I will gather the output of an sosreport"
+          
+  mkdir $EC2RL_GATHEREDDIR/supportconfig
+  supportconfig -R $EC2RL_GATHEREDDIR/supportconfig
+  echo "I have attempted to gather unsanitized sosreport output to $EC2RL_GATHEREDDIR/sosreport"
```
Comment (Drudenhaus) [inline mod.d/supportconfig.yaml:44]: s/sosreport/supportconfig

File: mod.d/supportconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      printf "%0.s=" {1..80}
+      echo -e "\nERROR:	"$BASH_COMMAND" exited with an error on line ${BASH_LINENO[0]}"
+      exit 0
+  }
+  trap error_trap ERR
+
+  # read-in shared function
+  source functions.bash
+
+  echo "I will gather the output of an sosreport"
```
Comment (Drudenhaus) [inline mod.d/supportconfig.yaml:40]: s/sosreport/supportconfig

**Review by Drudenhaus — CHANGES_REQUESTED** [review]
supportconfig module has multiple sosreport references.

File: mod.d/supportconfig.yaml (hunk 3 above)
Comment (gregbdunn) [reply mod.d/supportconfig.yaml]: Fixed these ups

**Review by Drudenhaus — APPROVED** [review]

**Comment by gregbdunn** [comment]
Drudenhaus is dealing with some permissions related issues so he can't issue official approval, but considering this approved for all intents and purposes.

**Pull request merged**